primitives/consensus/common/src/metrics.rs

```diff
@@ -0,0 +1,39 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Metering tools for consensus
+
+use prometheus_endpoint::{register, U64, Registry, PrometheusError, Opts, CounterVec};
+
+/// generic prometheus metrics for common consensus functionality.
+#[derive(Clone)]
+pub(crate) struct Metrics {
+	pub import_queue_processed: CounterVec<U64>,
+}
+
+impl Metrics {
+	pub(crate) fn register(registry: &Registry) -> Result<Self, PrometheusError> {
+		Ok(Self {
+			import_queue_processed: register(
+				CounterVec::new(
+					Opts::new("import_queue_processed", "Blocks processed by import queue"),
```

Comment:
```suggestion
					Opts::new("import_queue_processed_total", "Blocks processed by import queue"),
```

As it is a counter, the convention is to postfix `total` to it. See https://prometheus.io/docs/practices/naming/ for details.